modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/product_options/ProductOptionSelect.js

```diff
@@ -273,91 +273,92 @@ const ProductOptionSelect = ({
+				channelId,
+				productId,
+				accountId,
+				Liferay.CommerceContext
+					? Liferay.CommerceContext.currency.currencyCode
+					: '',
+				minQuantity,
+				null,
+				currentSkuOptions
+			)
+				.then((cpInstance) => {
+					setSelectedProductOptionValue({
+						...selectedProductOptionValue,
+						productOptionValueId: valueArray[0],
+						skuId: cpInstance.id,
+					});
 
-				if (currentCPInstanceSkuOption) {
-					const curIndex = currentSkuOptions.findIndex(
+					const currentCPInstanceSkuOption = currentSkuOptions.filter(
 						(skuOption) =>
 							skuOption.skuOptionKey === productOption.key
-					);
-
-					currentSkuOptions[curIndex] = {
-						...currentCPInstanceSkuOption,
-						key: productOption.key,
-					};
-				}
-
-				setSkuOptionsAtomState({
-					...skuOptionsAtomState,
-					[skuOptionsKey]: currentSkuOptions,
-				});
-
-				cpInstance.skuOptions = currentSkuOptions;
-				cpInstance.skuId = parseInt(cpInstance.id, 10);
-
-				currentSkuId = cpInstance.skuId;
+					)[0];
 
-				setSelectedSkuId(cpInstance.skuId);
+					if (currentCPInstanceSkuOption) {
+						const curIndex = currentSkuOptions.findIndex(
+							(skuOption) =>
+								skuOption.skuOptionKey === productOption.key
+						);
 
-				const dispatchedPayload = {
-					cpInstance,
-					namespace,
-				};
+						currentSkuOptions[curIndex] = {
+							...currentCPInstanceSkuOption,
+							key: productOption.key,
+						};
+					}
 
-				Liferay.fire(
-					`${namespace}${CP_INSTANCE_CHANGED}`,
-					dispatchedPayload
-				);
-			})
-			.finally(() => {
-				if (isMounted()) {
-					setHasErrors(false);
 					setSkuOptionsAtomState({
 						...skuOptionsAtomState,
-						[errorsKey]: getSkuOptionsErrors(
-							false,
-							isFromMiniCart,
-							productOption,
-							skuOptionsAtomState
-						),
 						[skuOptionsKey]: currentSkuOptions,
-						updating: false,
 					});
 
-					setTimeout(() =>
-						Liferay.fire(`${namespace}${CP_OPTION_CHANGED}`, {
-							productOptionId: productOption.id,
-							productOptionValueId: valueArray[0],
-							skuId: currentSkuId,
-							skuOptions: currentSkuOptions,
-						})
+					cpInstance.skuOptions = currentSkuOptions;
+					cpInstance.skuId = parseInt(cpInstance.id, 10);
+
+					currentSkuId = cpInstance.skuId;
+
+					setSelectedSkuId(cpInstance.skuId);
+
+					const dispatchedPayload = {
+						cpInstance,
+						namespace,
+					};
+
+					Liferay.fire(
+						`${namespace}${CP_INSTANCE_CHANGED}`,
+						dispatchedPayload
 					);
-				}
-			});
+				})
+				.finally(() => {
+					if (isMounted()) {
+						setHasErrors(false);
+						setSkuOptionsAtomState({
+							...skuOptionsAtomState,
+							[errorsKey]: getSkuOptionsErrors(
+								false,
+								isFromMiniCart,
+								productOption,
+								skuOptionsAtomState
+							),
+							[skuOptionsKey]: currentSkuOptions,
+							updating: false,
+						});
+
+						setTimeout(() =>
```

Comment:
This is technical debt and discouraged as it's concurrent and not predictable behavior.

Evaluate introducing (or reusing, if there's one already) an effect that listens to `skuOptionsAtomState` or `errors` and, if a condition is satisfied, `Liferay.fire`'s the `CP_OPTION_CHANGED` event.